File: api/ruby/filter_operators_by_availability.rb
#!/usr/bin/ruby

require 'rubygems'
require 'rest_client'
require 'json'

headers = {
  :authorization: 'Token MY_SECRET_API_TOKEN',
  :accept: 'application/vnd.salemove.v1+json'
}

def get_available_operators(operator_data)
  operator_data['operators'].select { |operator| operator['available'] }
end

response = RestClient.get 'https://api.salemove.com/operators', headers

operator_data = JSON.parse response

available_operators = get_available_operators operator_data

puts available_operators

File: api/javascript/filter_operators_by_availability.js
function request(method, url) {
  return new Promise(function(resolve, reject) {
    var xhr = new XMLHttpRequest();
    xhr.open(method, url);
    xhr.setRequestHeader('Authorization', 'Token MY_SECRET_API_TOKEN');
    xhr.setRequestHeader('Accept', 'application/vnd.salemove.v1+json');
    xhr.onload = function() {
      if (xhr.status >= 200 && xhr.status < 300) {
        resolve(xhr.responseText);
      } else {
        reject(xhr.statusText);
      }
    };
    xhr.onerror = function() { reject(xhr.statusText) };
    xhr.send();
  });
}

function filterAvailableOperators(data) {
  return data.operators.filter(function(operator) {
    return operator.available == true;
  });
};

request('GET', 'https://api.salemove.com/operators').then(function(result) {
  console.log(filterAvailableOperators(JSON.parse(result)));
}, function(error) {
  console.log('Error fetching data.');
});
